dbcog/monster_graph.py

```diff
@@ -590,6 +608,31 @@ def monster_is_black_medal_exchange(self, monster: MonsterModel) -> bool:
     def monster_is_black_medal_exchange_evo(self, monster: MonsterModel) -> bool:
         return any(self.monster_is_black_medal_exchange(alt) for alt in self.get_alt_monsters(monster))
 
+    def monster_is_currently_exchangable(self, monster: MonsterModel) -> bool:
+        models = self.get_monster_exchange_models(monster)
+        now = datetime.now()
+        for model in models:
+            if model.start_timestamp < now < model.end_timestamp:
+                return True
+        return False
+
+    def monster_is_currently_exchangable_evo(self, monster: MonsterModel) -> bool:
+        return any(self.monster_is_currently_exchangable(alt) for alt in self.get_alt_monsters(monster))
+
+    def monster_is_permanent_exchange(self, monster: MonsterModel) -> bool:
+        return any(model.permanent or model.end_timestamp.year > 2030
+                   for model in self.get_monster_exchange_models(monster))
+
+    def monster_is_permanent_exchange_evo(self, monster: MonsterModel) -> bool:
+        return any(self.monster_is_permanent_exchange(alt) for alt in self.get_alt_monsters(monster))
+
+    def monster_is_temporary_exchange(self, monster: MonsterModel) -> bool:
```

Comment:
i dont think i want to rename this function but can you add a docstring explicitly calling out that this is NOT necessarily current